### PR TITLE
feat(types): forbid using default log fn when custom only used (#1998)

### LINKED ISSUE #1998: Forbid using default log levels when `useOnlyCustomLevels: true`
Currently, pino's TS declaration allows using default log levels when `useOnlyCustomLevels` is `true`. This following code will result in a runtime error:

```ts
import { pino } from 'pino';

const logger = pino({
  customLevels: { customInfo: 10 },
  useOnlyCustomLevels: true,
});

logger.info('123'); // TS should warn here ?
logger.customInfo('123'); // should work fine
```

Expected behavior:

```ts
import { pino } from 'pino';

const logger = pino({
  customLevels: { customInfo: 10 },
  useOnlyCustomLevels: true,
});

logger.info('123'); // TS errors here
logger.customInfo('123'); // works fine
```

### PR DESCRIPTION
**Pull request by chernodub**
Forbids using default log methods when `useOnlyCustomLevels` used. 

Only improves types, the actual behavior were not changed.

Closes #1998

### CODE REVIEW

File: test/types/pino.ts [hunk 1]
```diff
@@ -45,34 +45,50 @@ loggerMulti.info('test2')
...
+    // @ts-expect-error
+    loggerCustomLevel.error('test');
+    // @ts-expect-error
+    loggerCustomLevel.warn('test');
+    // @ts-expect-error
+    loggerCustomLevel.info('test');
+    // @ts-expect-error
+    loggerCustomLevel.debug('test');
+    // @ts-expect-error
+    loggerCustomLevel.trace('test');
```
Comment (mcollina) [inline test/types/pino.ts:92]: Please use [`expectError`](https://github.com/tsdjs/tsd?tab=readme-ov-file#expecterrort--anyexpression-t) instead

Comment (chernodub) [reply test/types/pino.ts]: Thanks, @mcollina. I've moved the type tests to tsd files and used `expectError` as recommended!

**Review by mcollina — APPROVED** [review]
lgtm

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**